app/common/auth/__init__.py

```diff
@@ -14,31 +16,52 @@
 )
 
 
-@auth_blueprint.route("/sign-in", methods=["GET", "POST"])
-def sign_in() -> ResponseReturnValue:
+@auth_blueprint.route("/request-a-link-to-sign-in", methods=["GET", "POST"])
+@auto_commit_after_request
+def request_a_link_to_sign_in() -> ResponseReturnValue:
     form = SignInForm()
     if form.validate_on_submit():
-        email = form.email_address.data
+        email = cast(str, form.email_address.data)
 
-        # TODO: all session stuff will be revisited as part of FSPT-334
-        session["email_address"] = email
+        user = get_or_create_user(email_address=email)
+
+        # TODO: read+validate redirect from query params
+        magic_link = interfaces.magic_link.create_magic_link(
+            user=user, redirect_to_path=url_for("platform.list_grants")
+        )
 
         notification_service.send_magic_link(
-            email,  # type: ignore[arg-type]
-            magic_link_url="https://magic-link-tbd",
-            magic_link_expires_at_utc=datetime.datetime.now(ZoneInfo("UTC")) + datetime.timedelta(minutes=15),
-            request_new_magic_link_url="https://new-magic-link-tbd",
+            email,
+            magic_link_url=url_for("auth.claim_magic_link", magic_link_code=magic_link.code, _external=True),
+            magic_link_expires_at_utc=magic_link.expires_at_utc,
+            request_new_magic_link_url=url_for("auth.request_a_link_to_sign_in", _external=True),
```

Comment:
I think all of this can now be extracted from the magic link in the helper but that doesn't need to be done in this PR, feels like something that be done in a cleanup at the end once all the mechanics are in

Reply:
I did _start_ doing this, then stopped. We could pass in the magic link object and read magic_link.user.email from it, and magic_link.expires_at_utc. But if we want to generate the two URLs then this method will only work if there's a corresponding flask app/request context, which feels like maybe a bit of state I wanted to keep out of here.

Again, only very lightly held, but that was the reason I didn't do this refactor yet.